lib/narn-lib.js

```diff
@@ -22,45 +39,69 @@ exports.detectYarn = function detectYarn(args) {
 
 exports.getYarnArgs = narnArgs => narnArgs;
 
-exports.getNpmArgs = narnArgs => {
-  const yarnArgs = minimist(narnArgs, { boolean: ["dev", "D"] });
+exports.getNpmArgs = (narnArgs, isPnpm = false) => {
+  const yarnArgs = minimist(narnArgs, {
+    boolean: ["dev", "D"],
+    alias: {
+      D: "dev"
+    }
+  });
   const yarnCommands = yarnArgs._;
   const yarnTarget = yarnCommands.length > 0 ? yarnCommands[0] : "install";
   const yarnSubCommands = yarnCommands.slice(1);
+  const narnSubArgs = narnArgs.slice(1);
+
   let npmTarget;
   let npmArgs;
   let result;
   switch (yarnTarget) {
     case "install":
       npmTarget = "install";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
       break;
     case "add":
       npmTarget = "install";
       const packages = yarnSubCommands.map(transformNarnPackageString);
-      const saveType = yarnArgs.dev || yarnArgs.D ? "--save-dev" : "--save";
+      const saveType = yarnArgs.dev ? "--save-dev" : "--save";
       npmArgs = [saveType, ...packages];
       break;
     case "remove":
       npmTarget = "uninstall";
-      npmArgs = yarnSubCommands;
+      npmArgs = narnSubArgs;
       break;
     case "init":
       npmTarget = "init";
-      npmArgs = [];
+      npmArgs = narnSubArgs;
+      break;
+    case "update":
+    case "upgrade":
+      npmTarget = "update";
+      npmArgs = narnSubArgs;
+      break;
+    case "list":
+    case "prune":
+    case "version":
+      // commands that don't need further changes
+      npmTarget = yarnTarget;
+      npmArgs = narnSubArgs;
```

Comment:
Not all flags are the same between yarn and npm. We should not assume that they are

Reply:
I repeat, passing them around doesn't kill it's functionality, if you want to improve the flags adaptation on every scenario is possible, but right now this PR is not indended to do that.

Reply:
https://github.com/joeldenning/narn/pull/13#discussion_r376681334